### PR TITLE
Do not pass cached objects to inx workers to avoid leaks

### PR DESCRIPTION
**Pull request by muXxer**

### CODE REVIEW

File: plugins/inx/server_blocks.go [hunk 1]
```diff
@@ -184,13 +185,22 @@ func (s *Server) ListenToSolidBlocks(_ *inx.NoParams, srv inx.INX_ListenToSolidB
...
+			Plugin.LogInfof("send error: %v", err)
+			cancel()
+		}
+
+	}, workerpool.WorkerCount(workerCount), workerpool.QueueSize(workerQueueSize), workerpool.FlushTasksAtShutdown(true))
+
+	onBlockSolid := events.NewClosure(func(blockMeta *storage.CachedMetadata) {
 		defer blockMeta.Release(true) // meta -1
 
 		payload, err := NewINXBlockMetadata(ctx, blockMeta.Metadata().BlockID(), blockMeta.Metadata())
```
Comment (alexsporn) [inline plugins/inx/server_blocks.go:206]: the error log needs to be updated here, it is no send error

File: plugins/inx/server_blocks.go [hunk 2]
```diff
@@ -231,13 +233,22 @@ func (s *Server) ListenToReferencedBlocks(_ *inx.NoParams, srv inx.INX_ListenToR
...
+			Plugin.LogInfof("send error: %v", err)
+			cancel()
+		}
+
+	}, workerpool.WorkerCount(workerCount), workerpool.QueueSize(workerQueueSize), workerpool.FlushTasksAtShutdown(true))
+
+	onBlockReferenced := events.NewClosure(func(blockMeta *storage.CachedMetadata, index iotago.MilestoneIndex, confTime uint32) {
 		defer blockMeta.Release(true) // meta -1
 
 		payload, err := NewINXBlockMetadata(ctx, blockMeta.Metadata().BlockID(), blockMeta.Metadata())
```
Comment (alexsporn) [inline plugins/inx/server_blocks.go:254]: same here, no send error

File: plugins/inx/server_milestones.go [hunk 3]
```diff
@@ -28,7 +28,19 @@ func cachedMilestoneFromRequestOrNil(req *inx.MilestoneRequest) *storage.CachedM
...
+			ms.MilestoneID(),
+			ms.Index(),
+			ms.TimestampUnix()),
+		Milestone: &inx.RawMilestone{
+			Data: ms.Data(),
+		},
+	}
+}
+
+func inxMilestoneForCachedMilestone(ms *storage.CachedMilestone) *inx.Milestone {
```
Comment (alexsporn) [inline plugins/inx/server_milestones.go:43]: the other funcs don't have an `inx` prefix

File: plugins/inx/server_milestones.go [hunk 4]
```diff
@@ -28,7 +28,19 @@ func cachedMilestoneFromRequestOrNil(req *inx.MilestoneRequest) *storage.CachedM
 	return deps.Storage.CachedMilestoneByIndexOrNil(msIndex)
 }
 
-func milestoneForCachedMilestone(ms *storage.CachedMilestone) (*inx.Milestone, error) {
+func inxMilestoneForMilestone(ms *storage.Milestone) *inx.Milestone {
```
Comment (alexsporn) [inline plugins/inx/server_milestones.go:31]: the other funcs don't have an `inx` prefix

Comment (muXxer) [reply plugins/inx/server_milestones.go:31]: so milestoneForMilestone? :sweat_smile: 
eyeForAnEye :trollface:

Comment (alexsporn) [reply plugins/inx/server_milestones.go:31]: `milestoneForStoredMilestone` 🤷

**Review by alexsporn — APPROVED** [review]

**Pull request merged**